src/internal/types.ts

```diff
@@ -24,25 +34,57 @@ export type Mapped<T extends IterableContainer, K> = {
 export type IterableContainer<T = unknown> = ReadonlyArray<T> | readonly [];
 
 /**
- * We define a Simple record as one that doesn't define it's properties beyond
- * their general type. This is the case for string and number keys (and their
- * combination). Another way of thinking about simple records is that the number
- * of properties are unbound (they can be empty, or they can have 1000 props).
+ * Check if a type is guaranteed to be a bounded record: a record with a fixed
+ * set of keys.
  *
- * TODO: Template string literals (e.g. `prefix_${number}`) should also be
- * considered simple, but we don't know how to check for them...
+ * For example, a record with a union of string or number keys is unbounded,
```

Comment:
1. Would this be better as a JSDoc `@example`? 
2. Would this be clearer in code vs words (e.g. `Record<string, unknown>`)
3. Why start with a complex example (union) instead of the trivial one?